app/src/main/java/com/example/myapplication/ui/MainActivityCompose.kt

```diff
@@ -0,0 +1,133 @@
+package com.example.myapplication.ui
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color.Companion.Black
+import androidx.compose.ui.graphics.Color.Companion.Blue
+import androidx.compose.ui.graphics.Color.Companion.Cyan
+import androidx.compose.ui.graphics.Color.Companion.Gray
+import androidx.compose.ui.graphics.Color.Companion.Magenta
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MainActivityCompose : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val mainViewModel: MainActivityViewModel = viewModel()
+
+            FlexibleGird(
+                itemsCount = 10,
+                columnsCount = 3,
+                itemsHeight = 100,
+                gridWidth = 300,
+                itemsHeightPercent = 20,
+                parentPaddingStart = 15,
+                parentPaddingEnd = 15,
+                parentPaddingTop = 15,
+                parentPaddingBottom = 15,
+                itemDecorationVertical = 15,
+                itemDecorationHorizontal = 15,
+                hasDinamycHeight = false
+            )
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun DefaultPreview() {
+    FlexibleGird(
```

Comment:
Please add model entity here (GridViewState)
```suggestion
    FlexibleGird(
        val viewState: GridViewState
```
Where GridViewState is:
```suggestion
    data class GridViewState(
        val items: List<ItemViewState>, 
        val columnsCount: Int,
        val itemsHeight: ItemsHeight
    )
```
Where ItemsHeight is:
```suggestion
    sealed interface ItemsHeight {
        val heightPercent: Int
    
        data class Fixed(override val heightPercent: Int, val height: Int) : ItemsHeight
        data class Dynamic(override val heightPercent: Int) : ItemsHeight
    }
```

Reply:
Done